DataConnectors/SalesforceServiceCloud/SalesforceSentinelConnector/__init__.py

```diff
@@ -0,0 +1,204 @@
+import datetime
+import logging
+import json 
+import requests
+import hashlib
+import hmac
+import base64
+import csv
+import os
+import sys
+import tempfile
+import azure.functions as func
+
+
+customer_id = os.environ['WorkspaceID'] 
+shared_key = os.environ['WorkspaceKey']
+log_type = 'SalesforceServiceCloud'
+user = os.environ['SalesforceUser']
+password = os.environ['SalesforcePass']
+security_token = os.environ['SalesforceSecurityToken']
+consumer_key = os.environ['SalesforceConsumerKey']
+consumer_secret = os.environ['SalesforceConsumerSecret']
+object =  "EventLogFile"
+interval = "hourly"
+hours_interval = 1
+days_interval = 1
+url = "https://login.salesforce.com/services/oauth2/token"
+
+
+def _get_token():
+    params = {
+        "grant_type": "password",
+        "client_id": consumer_key,
+        "client_secret": consumer_secret,
+        "username": user,
+        "password": f'{password}{security_token}'
+    }
+    try:
+        r = requests.post(url, params=params)
+        _token = json.loads(r.text)['access_token']
+        _instance_url = json.loads(r.text)['instance_url']
+        return _token,_instance_url
+    except Exception as err:
+        logging.error(f'Token getting failed. Exiting program. {err}')
+        exit()
+
+
+def generate_date():
+    if interval == 'hourly':
+        current_time = datetime.datetime.utcnow().replace(second=0, microsecond=0)
+        past_time = current_time - datetime.timedelta(hours=hours_interval)
+    elif interval == 'daily':
+        current_time = datetime.datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+        past_time = current_time - datetime.timedelta(days=days_interval, hours=1)
+    return past_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def pull_log_files():
+    past_time = generate_date()
+
+    if interval == 'hourly':
+        query = "/services/data/v44.0/query?q=SELECT+Id+,+EventType+,+Interval+,+LogDate+,+LogFile+,+LogFileLength" + \
+        "+FROM+EventLogFile" + \
+        f"+WHERE+Interval+=+'Hourly'+and+CreatedDate+>+{past_time}"
+
+    elif interval == 'daily':
+        query = "/services/data/v44.0/query?q=SELECT+Id+,+CreatedDate+,+EventType+,+LogDate+,+LogFile+,+LogFileLength" + \
+                "+FROM+EventLogFile" + \
+                f"+WHERE+LogDate+>+{past_time}"
+    try:
+        logging.info('Searching files last modified from {}'.format(past_time))
+        r = requests.get(f'{instance_url}{query}', headers=headers)
```

Comment:
According to salesforce documentation, the result may contain the next nextRecordsUrl, needs to keep retrieving record from the returned URL. 

https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api/resources_query.htm

If the query results are too large, the response contains the first batch of results and a query identifier in the nextRecordsUrl field of the response. The identifier can be used in an additional request to retrieve the next batch.

Reply:
As we’re getting events on hourly basis (not daily) we haven’t faced situation when this could happen. But to be on safe side we’ll add handling of nextRecordsUrl.

Reply:
We require all all the connectors to handle pagination properly, thank you for fixing.